### PR TITLE
fix: use smaller-batch extrapolation when DeepSeek-V4 attention cubic interpolation fails

### PR DESCRIPTION
**Pull request by YijiaZhao**
#### Overview:

  Two fixes for DeepSeek-V4-Flash in AIC: (1) make mixed-batch context-attention                                                                                                                                                                    
  queries robust when the requested `(bs, isl)` falls outside the captured perf
  grid; (2) point the collector's V4-Flash default at the FP8 quant variant we                                                                                                                                                                      
  actually run on Hopper (H20), so the `deepseek-ai/DeepSeek-V4-Flash` logical                                                                                                                                                                      
  name no longer (silently) triggers any data collection on its own.                                                                                                                                                                                
                                                                                                                                                                                                                                                    
  #### Details:                                                                                                                                                                                                                                     
                                                                                                                                                                                                                                                    
  **1. Smaller-batch fallback in `_dsv4_flash_robust_3d_lookup`**                                                                                                                                                                                   
     (`src/aiconfigurator/sdk/perf_database.py`)
                                                                                                                                                                                                                                                    
  V4-Flash sweeps only b ∈ {1, 2, 4, 8, ..., 1024} and the grid is trapezoidal:
  larger b -> smaller max captured s. A query for an unsampled batch (e.g. bs=3                                                                                                                                                                     
  from a 3-request prefill batch with avg_isl=2670) hits a region where both                                                                                                                                                                        
  cubic and linear 3D interp return NaN (or raise on degenerate point clouds),                                                                                                                                                                      
  leaving the caller without a value.                                                                                                                                                                                                               
                                                                                                                                                                                                                                                    
  The robust lookup now adds a final fallback: pick the largest captured b' < z,                                                                                                                                                                    
  1D-interpolate along s at b' if exact y is missing, and linearly scale the                                                                                                                                                                        
  leaf by z/b'. FlashMLA varlen attention work scales near-linearly with batch                                                                                                                                                                      
  at fixed seq_len, so this is a physically reasonable extrapolation.                                                                                                                                                                               
                                                                                                                                                                                                                                                    
  **2. Collector V4-Flash default → sgl-project FP8**                                                                                                                                                                                               
     (`collector/common_test_cases.py`, `collector/collect.py`)                                                                                                                                                                                     
   
  `_DSV4_FLASH_DEFAULT_MODEL` and `collect.py`'s auto-expand trigger both move                                                                                                                                                                      
  from `deepseek-ai/DeepSeek-V4-Flash` (native MXFP4, Hopper unsupported) to                                                                                                                                                                        
  `sgl-project/DeepSeek-V4-Flash-FP8` (FP8 block quant, the variant we run).                                                                                                                                                                        
  After this change `--model-path deepseek-ai/DeepSeek-V4-Flash` is a no-op for                                                                                                                                                                     
  dsv4_flash_* collection — that name will be wired up later, separately.                                                                                                                                                                           
                                                                                                                                                                                                                                                    
  #### Where should the reviewer start?                                                                                                                                                                                                             
                                                                                                                                                                                                                                                    
  - `src/aiconfigurator/sdk/perf_database.py` — the fallback logic in
    `_dsv4_flash_robust_3d_lookup`. The hot-path early return for cubic/linear                                                                                                                                                                      
    is unchanged; only the previously-thrown failure case now extrapolates.                                                                                                                                                                         
  - `tests/unit/sdk/database/test_deepseek_v4_module.py` — two new tests                                                                                                                                                                            
    document the fallback contract (interp-along-s + scale-by-batch, plus the                                                                                                                                                                       
    trapezoidal case where the closest b' has no anchor at the query s).                                                                                                                                                                            
  - The two collector-side one-liners are mechanical.    

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

* **Bug Fixes**
  * Updated V4-Flash model configuration for improved system compatibility.
  * Enhanced performance lookup with robust fallback strategies for better edge case handling.

* **Tests**
  * Added comprehensive test coverage for V4-Flash performance lookup edge cases.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

### CODE REVIEW

File: src/aiconfigurator/sdk/perf_database.py [hunk 1]
```diff
@@ -1374,10 +1374,35 @@ def _dsv4_flash_robust_3d_lookup(self, dict_, x, y, z):
...
-    except Exception:
-        return self._interp_3d(x, y, z, dict_, "linear")
+    import math
+
+    for m in ("cubic", "linear"):
+        try:
+            r = self._interp_3d(x, y, z, dict_, m)
+            v = r.get("latency") if isinstance(r, dict) else None
+            if v is not None and not (isinstance(v, float) and math.isnan(v)):
+                return r
```
[the remark this answers is not in the document]
Comment (YijiaZhao) [reply src/aiconfigurator/sdk/perf_database.py]: fixed

File: src/aiconfigurator/sdk/perf_database.py [hunk 2]
```diff
@@ -1374,10 +1374,35 @@ def _dsv4_flash_robust_3d_lookup(self, dict_, x, y, z):
...
+            try:
+                if y in sub and bp in sub[y]:
+                    leaf = sub[y][bp]
+                elif len(ss) >= 2:
+                    sl, sr = self._nearest_1d_point_helper(y, ss, inner_only=False)
+                    leaf = self._interp_1d([sl, sr], [sub[sl][bp], sub[sr][bp]], y)
+                else:
+                    continue
+                return {f: leaf.get(f, 0.0) * z / bp for f in ("latency", "power", "energy")}
+            except Exception:
```
[the remark this answers is not in the document]
Comment (YijiaZhao) [reply src/aiconfigurator/sdk/perf_database.py]: fixed

**Review by Arsene12358 — APPROVED** [review]
